src/AlgebraicGeometry/Schemes/ProjectiveVariety/Objects/Constructors.jl

```diff
@@ -0,0 +1,82 @@
+########################################################
+# (1) Generic constructors
+########################################################
+@doc raw"""
+    projective_variety(X::AbsProjectiveScheme; check::Bool=true) -> ProjectiveVariety
+
+Convert ``X`` to an projective variety.
+
+If check is set, then compute the reduced scheme of `X` first.
+"""
+function projective_variety(X::AbsProjectiveScheme{<:Field}; check::Bool=true)
+  check  ||  return ProjectiveVariety(X, check=check)
+  Xred,_ = reduced_scheme(X)
+  return ProjectiveVariety(Xred, check=check)
+end
+
+@doc raw"""
+    projective_variety(I::MPolyIdeal; check::Bool=true) -> ProjectiveVariety
+
+Return the projective variety defined by the homogeneous prime ideal ``I``.
+
+Since our varieties are irreducible, we check that ``I`` stays prime when
+viewed over the algebraic closure. This is an expensive check that can be disabled.
+"""
+projective_variety(I::MPolyIdeal{<:MPolyDecRingElem}; check::Bool=true) = ProjectiveVariety(ProjectiveScheme(base_ring(I),I), check=check)
+
+function projective_variety(R::MPolyDecRing, I::MPolyIdeal{<:MPolyDecRingElem}; check::Bool=true)
+  @req base_ring(I) === R "ideal must be defined over R"
+  ProjectiveVariety(ProjectiveScheme(R,I), check=check)
+end
+
+
+@doc raw"""
+    projective_variety(R::Ring; check::Bool=true)
+
+Return the projective variety defined by the ``\mathbb{Z}`` graded ring ``R``.
+
+We require that ``R`` is a finitely generated algebra over a field ``k`` and
+moreover that the base change of ``R`` to the algebraic closure ``\bar k``
+is an integral domain.
```

Comment:
Unfortunately, we also require that `R` is the quotient of a standard graded ring.

Reply:
what is a standard graded ring?

Reply:
Every variable has the same degree?

Reply:
Usually, this means ZZ-graded with all variables of degree 1

Reply:
Thanks. Where is this requirement coming from?  Are there intrinsic reasons? 
(I would imagine that a groebner basis of a homogeneous ideal of a non-standard graded ring is possibly not homogeneous? Or is it just a matter of picking a better monomial ordering?)

Reply:
So why are we restricting to standard gradings if the most important tool works?

Reply:
I have not written this code, so I can only guess: You probably want to define a projectice variety by entering its coordinate ring? If so, and if the variety is NOT STANDARD ZZ-graded, you need to consider this as a variety in a weighted projective space. Or, what else is meant?

Reply:
Exactly.

Reply:
Yes, that's it. Non-standard gradings are corresponding to weighted projective spaces. The proj-construction is different there when it comes to passing to the affine charts. We simply don't have that functionality yet.